akka-docs/src/main/paradox/typed/cluster-membership.md

```diff
@@ -58,60 +58,74 @@ Note that the node might already have been shutdown when this event is published
-state to `up`.
+A node is introduced to the cluster by invoking the `join` action which puts the node in the `joining` state.
+Once all nodes have seen that the new node is joining (through @ref[gossip convergence](cluster-concepts.md#gossip-convergence)) the
+@ref[`leader`](cluster-concepts.md#leader) will set the member state to `up`.
 
 If a node is leaving the cluster in a safe, expected manner, for example through @ref[coordinated shutdown](../coordinated-shutdown.md), 
-it switches to the `leaving` state. Once the leader sees the convergence on the node in the
+it invokes the `leave` action which switches it to the `leaving` state. Once the leader sees the convergence on the node in the
 `leaving` state, the leader will then move it to `exiting`.  Once all nodes
 have seen the exiting state (convergence) the `leader` will remove the node
 from the cluster, marking it as `removed`.
 
 If a node is `unreachable` then gossip convergence is not possible and therefore
-any `leader` actions are also not possible (for instance, allowing a node to
-become a part of the cluster). To be able to move forward the state of the
-`unreachable` nodes must be changed. It must become `reachable` again or marked
-as `down`. If the node is to join the cluster again the actor system must be
-restarted and go through the joining process again. If new incarnation of the unreachable
-node tries to rejoin the cluster old incarnation will be marked as `down` and new
-incarnation can rejoin the cluster without manual intervention. 
+most `leader` actions are impossible (for instance, allowing a node to
+become a part of the cluster). To be able to move forward, the node must become `reachable` again
+or the node must be explicitly "downed". This is required because the state of an unreachable node is unknown and the
+cluster cannot know if the node has crashed or is only temporarily unreachable because of network issues or GC pauses.
+See the section about [User Actions](#user-actions) below for ways a node can be downed.
+
+The actor system on a node that exited or was downed cannot join the cluster again. In particular, a node that was downed while being
+unreachable and then regains connectivity cannot rejoin the cluster. Instead, the process has to be restarted on the
+node, creating a new actor system that can go through the joining process again.
+
+A special case is a node that was restarted without going through the leaving or downing process e.g. because the machine
+hosting the node was unexpectedly restarted. When the new instance of the node tries to rejoin the cluster, the cluster
+might still track the old instance as unreachable. In this case, however, it is clear that the old node is gone because
+the new instance will have the same address (host and port) as its old instance. In this case, the previous instance
+will be automatically marked as `down` and the new instance can rejoin the cluster without manual intervention.
+
+## Leader
```

Comment:
I added this complete section to explain the leader and how it relates to state changes.